python-frontend/src/main/java/org/sonar/python/types/TypeShed.java

```diff
@@ -193,6 +149,87 @@ public static Symbol symbolWithFQN(String stdLibModuleName, String fullyQualifie
     return null;
   }
 
+  @CheckForNull
+  public static Symbol symbolWithFQN(String fullyQualifiedName) {
+    Map<String, Symbol> builtinSymbols = builtinSymbols();
+    Symbol builtinSymbol = builtinSymbols.get(normalizedFqn(fullyQualifiedName));
+    if (builtinSymbol != null) {
+      return builtinSymbol;
+    }
+    String[] fqnSplittedByDot = fullyQualifiedName.split("\\.");
+    String moduleName = Arrays.stream(fqnSplittedByDot, 0, fqnSplittedByDot.length - 1).collect(Collectors.joining("."));
+    return symbolWithFQN(moduleName, fullyQualifiedName);
+  }
+
+  /**
+   * Returns stub symbols to be used by SonarSecurity.
+   * Ambiguous symbols that only contain class symbols are disambiguated with latest Python version.
+   */
+  public static Collection<Symbol> stubFilesSymbols() {
+    Set<Symbol> symbols = new HashSet<>(TypeShed.builtinSymbols().values());
+    for (Map<String, Symbol> symbolsByFqn : typeShedSymbols.values()) {
+      for (Symbol symbol : symbolsByFqn.values()) {
+        symbols.add(isAmbiguousSymbolOfClasses(symbol) ? disambiguateWithLatestPythonSymbol(((AmbiguousSymbol) symbol).alternatives()) : symbol);
+      }
+    }
+    return symbols;
+  }
+
+  public static String normalizedFqn(String fqn) {
+    if (fqn.startsWith(BUILTINS_PREFIX)) {
+      return fqn.substring(BUILTINS_PREFIX.length());
+    }
+    return fqn;
+  }
+
+  public static boolean isValidForProjectPythonVersion(List<String> validForPythonVersions) {
+    if (validForPythonVersions.isEmpty()) {
+      return true;
+    }
+    HashSet<String> intersection = new HashSet<>(validForPythonVersions);
+    intersection.retainAll(supportedPythonVersions);
+    return !intersection.isEmpty();
+  }
+
+  public static Set<Symbol> symbolsFromDescriptor(Set<Object> descriptors, boolean isInsideClass) {
```

Comment:
Now that we have actual `Descriptor` classes, perhaps it would be better to rename this concept `ProtobufDescriptor` or something similar to clearly distinguish them.